### PR TITLE
Update tanka docs.

### PR DESCRIPTION
**Pull request by mplzik**
Fix/update the tanka docs.

Signed-off-by: Milan Plzik <milan.plzik@grafana.com>

### CODE REVIEW

File: docs/docs/tutorial/jsonnet.mdx [hunk 1]
```diff
@@ -286,12 +286,15 @@ clusters just work.
 This allows us to make sure that you will never accidentally apply to the wrong
 cluster.
 
-> **Note**: Tanka won't create the namespace for you. Either include it in
-> Jsonnet, or with the following command:
+> **Note**: Tanka won't create the namespace for you -- you need to include it in
+> Jsonnet. Alternatively, you can create the namespace manually:
```
Comment (malcolmholmes) [inline docs/docs/tutorial/jsonnet.mdx]: would be great to show how you *do* create a namespace via Jsonnet. It isn't particularly hard. I think this will do it:

```
{
  my_namespace: $.core.v1.namespace.new('my-namespace'),
}

File: docs/docs/tutorial/refresher.mdx [hunk 2]
```diff
@@ -17,8 +17,7 @@ for the task of deploying Grafana and Prometheus to Kubernetes:
      Prometheus.
 2. Grafana
    - Another `Deployment` is required for the Grafana server.
-   - To connect to the webinterface, we also need a `Service` of type
-     `NodePort`.
+   - To connect to the webinterface, we also need a `Service`.
```
Comment (sh0rez) [inline docs/docs/tutorial/refresher.mdx]: The idea with `NodePort` here was to be able to use a mixin later on: 

https://github.com/grafana/tanka/blob/b32282ab9220ad87c70d7f7b35c5f9daa3c4160e/docs/docs/tutorial/k-lib.mdx#L117-L118

Although I get this is a bad example as it is completely redundant

Comment (mplzik) [reply docs/docs/tutorial/refresher.mdx]: Hmm, gotcha. What about a wording, which puts less emphasis on the NodePort? That might just do the trick:

```
To connect to the webinterface, we will be using a `Service` of type `NodePort`.
```

File: docs/docs/tutorial/jsonnet.mdx (hunk 1 above)
Comment (mplzik) [reply docs/docs/tutorial/jsonnet.mdx]: I added the snippet to the docs.

File: docs/docs/faq.md [hunk 3]
```diff
@@ -35,6 +35,13 @@ first.
 Jsonnet on the other hand got you covered by supporting mixing (patching,
 deep-merging) objects on top of the libraries output if required.
 
+## What about kubecfg ?
+
+Tanka development has started at the time when kubecfg was a part of
+already-deprecated `ksonnet` project. Although these projects are similar, Tanka
+aims to provide continuity for `ksonnet` users, whereas `kubecfg`goes the way of
+a very simple jsonnet-to-kubernetes synchronization tool.
+
```
[the remark this answers is not in the document]
Comment (mplzik) [reply docs/docs/faq.md]: Reordered in the newest version.

File: docs/docs/tutorial/refresher.mdx (hunk 2 above)
Comment (mplzik) [reply docs/docs/tutorial/refresher.mdx]: This has been incorporated to the PR.

File: docs/docs/tutorial/jsonnet.mdx [hunk 4]
```diff
@@ -286,12 +286,22 @@ clusters just work.
 This allows us to make sure that you will never accidentally apply to the wrong
 cluster.
 
-> **Note**: Tanka won't create the namespace for you. Either include it in
-> Jsonnet, or with the following command:
+> **Note**: Tanka won't create the namespace for you -- you need to include it in
+> Jsonnet using:
+> ```json
+> {
+>   my_namespace: $.core.v1.namespace.new('my-namespace'),
```
Comment (sh0rez) [inline docs/docs/tutorial/jsonnet.mdx]: At this time ksonnet lib is not introduced yet .. so this snippet is not obvious and won’t work. 

We at least need a link I guess

Comment (mplzik) [reply docs/docs/tutorial/jsonnet.mdx]: Slightly updated, PTAL.

File: docs/docs/tutorial/jsonnet.mdx [hunk 5]
```diff
@@ -286,12 +286,23 @@ clusters just work.
 This allows us to make sure that you will never accidentally apply to the wrong
 cluster.
 
-> **Note**: Tanka won't create the namespace for you. Either include it in
-> Jsonnet, or with the following command:
+> **Note**: Tanka won't create the namespace for you -- you need to include it in
+> Jsonnet. Using ksonnet lib (mentioned in [Kubernetes library](/tutorial/k-lib)
+> section), the following snippet will take care of the namespace creation:
+> ```json
```
Comment (sh0rez) [inline docs/docs/tutorial/jsonnet.mdx]:
```suggestion
> ```jsonnet
```

We have support for that, much better syntax highlighting

**Comment by mplzik** [comment]
PR updated to only show namespace updates via raw kubernetes objects until ksonnet lib has been introduced; PTAL.

File: docs/docs/tutorial/jsonnet.mdx [hunk 6]
```diff
@@ -286,12 +286,28 @@ clusters just work.
 This allows us to make sure that you will never accidentally apply to the wrong
 cluster.
 
-> **Note**: Tanka won't create the namespace for you. Either include it in
-> Jsonnet, or with the following command:
+> **Note**: Tanka won't create the namespace for you -- you need to include it in
+> Jsonnet:
+> ```jsonnet
+> {
+>   "my_namespace": {
```
Comment (sh0rez) [inline docs/docs/tutorial/jsonnet.mdx]: This is Jsonnet, keys don‘t need quoting

Comment (mplzik) [reply docs/docs/tutorial/jsonnet.mdx]: Fixed. To be honest, I was lazy here, and just grabbed YAML namespace definition and converted it via a YAML to JSON converter O:-) . Forgot about jsonnet details.

File: docs/docs/jsonnet/overview.md [hunk 7]
```diff
@@ -10,7 +10,7 @@ menu: Writing Jsonnet
 expressing what shall be deployed to your Kubernetes cluster. Understanding
 Jsonnet is crucial to using Tanka effectively.
 
-This page covers the Jsonnet language itself. For more information on how to use Jsonnet with Kubernetes, see [the tutorial](/tutorial/jsonnet).
+This page covers the Jsonnet language itself. For more information on how to use Jsonnet with Kubernetes, see [the tutorial](/tutorial/jsonnet). For more information on Jsonnet itself, please have a look at [the official Jsonnet tutorial](https://jsonnet.org/learning/tutorial.html)
```
Comment (sh0rez) [inline docs/docs/jsonnet/overview.md]: This sentence is pretty duplicated .. maybe reword it a bit? Right now it reads like 

... Jsonnet language itself ... For more information ... For more information ... Jsonnet itself

Comment (mplzik) [reply docs/docs/jsonnet/overview.md]: Yup, fixed in the most recent commit. PTAL.

File: docs/docs/faq.md [hunk 8]
```diff
@@ -20,6 +34,13 @@ designing Tanka, we felt these add more complexity for the user than they
 provide additional value. To keep Tanka as minimal as possible, these are **not
 available** and are not likely to be ever added.
 
+## What about kubecfg ?
+
+Tanka development has started at the time when kubecfg was a part of
+already-deprecated `ksonnet` project. Although these projects are similar, Tanka
+aims to provide continuity for `ksonnet` users, whereas `kubecfg`goes the way of
```
Comment (sh0rez) [inline docs/docs/faq.md]:
```suggestion
aims to provide continuity for `ksonnet` users, whereas `kubecfg` goes the way of
```

File: docs/docs/faq.md [hunk 9]
```diff
@@ -20,6 +34,13 @@ designing Tanka, we felt these add more complexity for the user than they
...
+a very simple jsonnet-to-kubernetes synchronization tool.
```
Comment (sh0rez) [inline docs/docs/faq.md]: Is this their official project goal? We might underrate their efforts on the project in an potentially offending way.

Comment (mplzik) [reply docs/docs/faq.md]: Fair enough, might sound that way (although that wasn't the intent -- I meant it more in the "take jsonnet with no extra sugaring and apply it to the cluster). To be more correct, I'll use the phrase from kubecfg's `README.md`:
```
...is really just a thin Kubernetes-specific wrapper around jsonnet evaluation
```

Comment (mplzik) [reply docs/docs/faq.md]: Fixed.

File: docs/docs/faq.md [hunk 10]
```diff
@@ -20,6 +34,14 @@ designing Tanka, we felt these add more complexity for the user than they
 provide additional value. To keep Tanka as minimal as possible, these are **not
 available** and are not likely to be ever added.
 
+## What about kubecfg ?
+
+Tanka development has started at the time when kubecfg was a part of
+already-deprecated `ksonnet` project. Although these projects are similar, Tanka
+aims to provide continuity for `ksonnet` users, whereas `kubecfg` is (according
+to project's [README.md](https://github.com/bitnami/kubecfg/blob/master/README.md)
```
Comment (sh0rez) [inline docs/docs/faq.md]:
```suggestion
to the project's [README.md](https://github.com/bitnami/kubecfg/blob/master/README.md)
```
